sale_rental/models/stock_warehouse.py

```diff
@@ -0,0 +1,251 @@
+                        ("company_id", "=", self.company_id.id),
+                    ],
+                    limit=1,
+                )
+                if not out_loc:
+                    out_loc = slo.with_context(lang="en_US").create(
+                        {
+                            "name": "Rental Out",
+                            "location_id": wh.rental_view_location_id.id,
+                            "company_id": self.company_id.id,
+                        }
+                    )
+                    slo.browse(out_loc.id).name = _("Rental Out")
+                    logger.debug(
+                        "New out rental stock location created ID %d", out_loc.id
+                    )
+                wh.rental_out_location_id = out_loc.id
+
+    def write(self, vals):
+        if "rental_allowed" in vals:
+            rental_route = self.env.ref("sale_rental.route_warehouse0_rental")
+            sell_rented_route = self.env.ref(
+                "sale_rental.route_warehouse0_sell_rented_product"
+            )
+            if vals.get("rental_allowed"):
+                self._create_rental_locations()
+                self.write(
+                    {
+                        "route_ids": [(4, rental_route.id)],
+                        "rental_route_id": rental_route.id,
+                        "sell_rented_product_route_id": sell_rented_route.id,
+                    }
+                )
+                rental_rules = self.env["stock.rule"].search(
+                    [
+                        ("route_id", "in", [rental_route.id, sell_rented_route.id]),
+                        ("active", "=", False),
+                    ]
+                )
+                if rental_rules:
+                    rental_rules.write({"active": True})
+                else:
+                    for rule_vals in self._get_rental_push_pull_rules():
+                        self.env["stock.rule"].create(rule_vals)
+            else:
+                for wh in self:
+                    rules_to_archive = self.env["stock.rule"].search(
+                        [
+                            (
+                                "route_id",
+                                "in",
+                                (
+                                    wh.rental_route_id.id,
+                                    wh.sell_rented_product_route_id.id,
+                                ),
+                            ),
+                            ("company_id", "=", wh.company_id.id),
+                        ]
+                    )
+                    rules_to_archive.write({"active": False})
+                    wh.write(
+                        {
+                            "route_ids": [(3, rental_route.id)],
+                            "rental_route_id": False,
+                            "sell_rented_product_route_id": False,
+                        }
+                    )
```

Comment:
This should be done in function create as well.
Otherwise if you create a new warehouse, setting rental_allowed to true the locations / rules will not be created and routes will not be assigned.